examples/campaign_management/add_complete_campaigns_using_batch_job.rb

```diff
@@ -0,0 +1,370 @@
+    mutate_operations: build_all_operations(client, customer_id),
+  )
+  puts "#{response.total_operations} mutate operations have been added so far."
+
+  # You can use this next sequence token for calling
+  # add_all_batch_job_operations() next time
+  puts "Next sequence token for adding next operations is " \
+    "'#{response.next_sequence_token}'"
+end
+
+# Requests the API to run the batch job for executing all uploaded batch job
+# operations.
+def run_batch_job(batch_job_service_client, batch_job_resource_name)
+  operation_response = batch_job_service_client.run_batch_job(
+    resource_name: batch_job_resource_name,
+  )
+  puts "Batch job with resource name '#{batch_job_resource_name}' " \
+    "has been executed."
+  operation_response
+end
+
+# Polls the server until the batch job execution finishes by setting the initial
+# poll delay time and the total time to wait before time-out.
+def poll_batch_job(operation_response)
+  operation_response.wait_until_done!
+end
+
+# Prints all the results from running the batch job.
+def fetch_and_print_results(batch_job_service_client, batch_job_resource_name)
+  puts "Batch job with resource name '#{batch_job_resource_name}' has " \
+    "finished. Now, printing its results..." \
+
+  # Gets all the results from running batch job and print their information.
+  batch_job_results = batch_job_service_client.list_batch_job_results(
+    resource_name: batch_job_resource_name,
+    page_size: PAGE_SIZE,
+  )
+  batch_job_results.each do |result|
+    puts "Batch job ##{result.operation_index} has a status " \
+      "#{result.status ? result.status.message : 'N/A'} and response of type " \
+      "#{result.mutate_operation_response ? result.mutate_operation_response.response : 'N/A'}"
+  end
+end
+
+# Builds all operations for creating a complete campaign and return an array of
+# their corresponding mutate operations.
+def build_all_operations(client, customer_id)
+  mutate_operations = []
+
+  # Creates a new campaign budget operation and add it to the array of mutate
+  # operations.
+  campaign_budget_operation = build_campaign_budget_operation(client, customer_id)
+  mop = client.operation.mutate
+  mop.campaign_budget_operation = campaign_budget_operation
+  mutate_operations << mop
+
+  # Creates new campaign operations and adds them to the array of mutate
+  # operations.
+  campaign_operations = build_campaign_operations(
+    client, customer_id, campaign_budget_operation.create.resource_name)
+  campaign_operations.each do |op|
+    mop = client.operation.mutate
+    mop.campaign_operation = op
+    mutate_operations << mop
+  end
+
+  # Creates new campaign criterion operations and adds them to the array of
+  # mutate operations.
+  campaign_criterion_operations = build_campaign_criterion_operations(
+    client, campaign_operations)
+  campaign_criterion_operations.each do |op|
+    mop = client.operation.mutate
+    mop.campaign_criterion_operation = op
```

Comment:
Does "client.operation.mutate" not support a block like "client.resource.whatever" would? If it does support that, please use that syntax. If it doesn't, please file an issue for our library to add that functionality and then update this example to use that.

Reply:
It does not support block syntax. I'll file an issue to add that and update example later.